Jenkinsfile

```diff
@@ -3,767 +3,925 @@
-    DEV_SUFFIX = "dev"
-
-    // TEST_PROJECT contains the test deployment. The test image is a candidate for promotion to prod.
-    TEST_PROJECT = "moe-gwells-test"
-    TEST_SUFFIX = "staging"
-    TEST_HOST = "gwells-test.pathfinder.gov.bc.ca"
-
-    // PROD_PROJECT is the prod deployment.
-    // New production images can be deployed by tagging an existing "test" image as "prod".
-    PROD_PROJECT = "moe-gwells-prod"
-    PROD_SUFFIX = "production"
-    PROD_HOST = "gwells-prod.pathfinder.gov.bc.ca"
-
-    // PR_NUM is the pull request number e.g. 'pr-4'
-    PR_NUM = "${env.JOB_BASE_NAME}".toLowerCase()
-  }
-  agent any
-  stages {
-
-    // the Start Pipeline stage will process and apply OpenShift build templates which will create
-    // buildconfigs and an imagestream for built images.
-    // each pull request gets its own buildconfig but all new builds are pushed to a single imagestream,
-    // to be tagged with the pull request number.
-    // e.g.:  gwells-app:pr-999
-    stage('Prepare Templates') {
-      steps {
-        script {
-          echo "Cancelling previous builds..."
-          timeout(10) {
-            abortAllPreviousBuildInProgress(currentBuild)
-          }
-          echo "Previous builds cancelled"
-
-          openshift.withCluster() {
-            openshift.withProject(TOOLS_PROJECT) {
-
-              // Process db and app template into list objects
-              //  - variable substitution
-              echo "Processing build templates"
-              def dbtemplate = openshift.process("-f",
-                "openshift/postgresql.bc.json",
-                "ENV_NAME=${DEV_SUFFIX}"
-              )
-              //
-              def buildtemplate = openshift.process("-f",
-                "openshift/backend.bc.json",
-                "ENV_NAME=${DEV_SUFFIX}",
-                "NAME_SUFFIX=-${DEV_SUFFIX}-${PR_NUM}",
-                "APP_IMAGE_TAG=${PR_NUM}",
-                "SOURCE_REPOSITORY_URL=${REPOSITORY}",
-                "SOURCE_REPOSITORY_REF=pull/${CHANGE_ID}/head"
-              )
-
-              // Apply oc list objects
-              //  - add docker image reference as tag in gwells-postgresql
-              echo "Preparing database imagestream"
-              echo " \$ oc process -f openshift/postgresql.bc.json -p ENV_NAME=${DEV_SUFFIX} | oc apply -n moe-gwells-tools -f -"
-              openshift.apply(dbtemplate)
-              //  - add docker image reference as tag in gwells-application
-              //  - create build config
-              echo "Preparing backend imagestream and buildconfig"
-              echo " \$ oc process -f openshift/backend.bc.json -p ENV_NAME=${DEV_SUFFIX} -p NAME_SUFFIX=-${DEV_SUFFIX}-${PR_NUM} -p APP_IMAGE_TAG=${PR_NUM} -p SOURCE_REPOSITORY_URL=${REPOSITORY} -p SOURCE_REPOSITORY_REF=pull/${CHANGE_ID}/head | oc apply -n moe-gwells-tools -f -"
-              openshift.apply(buildtemplate)
+        REPOSITORY = 'https://www.github.com/bcgov/gwells.git'
```

Comment:
Maybe a stupid question, and definitely not to be addressed in the context of this code review - but couldn't we be getting these settings from outside configuration rather than having it baked straight into the jenkins file? It would make things a bit neater.

Reply:
@Sybrand That's on the roadmap as part of the modular pipeline.  We'd use a separate config or environment variables.